### ISSUE TITLE
128k context support?

### ISSUE THREAD

**Issue by wizd** [oobabooga/text-generation-webui#4454]
**Description**

New long-context models have emerged, such as [Yarn-Mistral-7b-128k](https://huggingface.co/NousResearch/Yarn-Mistral-7b-128k), but the current text generation web UI only supports 32k.

**Additional Context**

I hope to provide support for longer contexts.

**Comment by DutchEllie** [comment]
As far as I know this has already been added into llama.cpp by itself like a few days ago. So for gguf models it should work already.

**Comment by Ph0rk0z** [comment]
Have any of you actually attempted to run 128K? Or even the full 32k?

**Comment by Belarrius1** [comment]
Yup. "LoneStriker_Yarn-Mistral-7b-128k-8.0bpw-h8-exl2" it's works well with 32k in oobabooga text generation webui with I can't test with 128k because of UI ilimited to 32k. Wait and see

**Comment by berkut1** [comment]
I wonder how much VRAM you need to run 128k (in a fully 128k context)? For example, the 7b (Q_6 mistral model) with 8k (~28 000 symbols of context) requires ~11 GB of VRAM (6GB model + 4GB context).

**Comment by Ph0rk0z** [comment]
Oh by your math, about 64gb of vram.

**Comment by berkut1** [comment]
@Ph0rk0z that sounds like I'm wrong :D 
But with my 2080 GTX, I can run mistral with fully 8k context only, if allocate ~12 GPU layers, instead of 35. This is probably how CUDA works, because when I run the model without VRAM, it doesn't use as much RAM for context.

**Comment by Ph0rk0z** [comment]
He's getting about 2gb per 2048 by his math. There are things like flash attention, sliding window, 8bit cache, etc. I know for 32k on a l-1 33b it took all 48gb in 4bits-32g. One of us has to load it up and fill it.. I mostly chat so after 4-8k the wait time gets annoying while adding very little and often reducing quality.

**Comment by Belarrius1** [comment]
We have 200k model context now too

**Comment by mongolu** [comment]
> We have 200k model context now too

Try editing the html values for now

**Comment by esherriff** [comment]
This issue is absolutely still relevant, with increasing numbers of fine tunes of Yi-200k and a number of YaRN scaled models available, the UI slider should not be restricted to 32k context.
